File: app/src/main/java/com/ikem/nwodo/cryptonyte/NumberFormatter.java
package com.ikem.nwodo.cryptonyte;

import java.text.DecimalFormat;
import java.text.NumberFormat;
import java.util.Locale;

public class NumberFormatter {

    public static String currencyFormatter(long input){
        NumberFormat nf = NumberFormat.getCurrencyInstance(Locale.US);
        return nf.format(input);
    }

    public static String graphYAxisFormatter(float input){
        NumberFormat nf = NumberFormat.getInstance();

        return nf.format((int)input);
    }

}

File: app/src/main/java/com/ikem/nwodo/cryptonyte/utils/graph/NumberFormatter.kt
package com.ikem.nwodo.cryptonyte.utils.graph

import java.text.DecimalFormat
import java.text.NumberFormat
import java.util.*


object NumberFormatter {

    fun currencyFormatter(input: Long): String {
        val nf = NumberFormat.getCurrencyInstance(Locale.US)
        return nf.format(input)
    }

    fun graphYAxisFormatter(input: Float): String {
        val nf = NumberFormat.getInstance()
        return nf.format(input.toInt().toLong())
    }

    @JvmStatic
    fun priceFormatter(input: String?): String {
        val nf = DecimalFormat("##,###.00")
        var result = ""
        if (input != null)
            result = nf.format(input.toDouble())

        return result
    }

    @JvmStatic
    fun coinNameFormatter(input: String?): String {
        var result = ""
        if (input != null) result = (if (input.contains(' ')) {
            input.replace(' ', '\n')
        } else {
            input
        }).toString()
        return result
    }
}
